### PR TITLE
LPS-145528 Adapt background icon and text depending on allowed file types on item selector

### PR DESCRIPTION
**Pull request by boton**
Issue: https://issues.liferay.com/browse/LPS-145528

Add the `getMimeTypeRestriction` to ItemSelectorCriterion to update the new single file uploader, based on allowed types.

#### Default
image

#### Image
image
image

#### Video
image
image

### CODE REVIEW

**Comment by boton** [comment]
ci:test:sf

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
I'm not sure if [JournalItemSelectorCriterion](https://github.com/liferay/liferay-portal/blob/27a8205f095df4b630ca05501681f1c24e8b1f37/modules/apps/journal/journal-item-selector-api/src/main/java/com/liferay/journal/item/selector/criterion/JournalItemSelectorCriterion.java) is used only for images. 

/cc @adolfopa @AliciaGarciaGarcia

**Comment by boton** [comment]
ci:test:sf

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/blogs/blogs-item-selector-api/src/main/java/com/liferay/blogs/item/selector/criterion/BlogsItemSelectorCriterion.java [hunk 1]
```diff
@@ -15,9 +15,16 @@
...
 
 /**
  * @author Roberto Díaz
  */
 public class BlogsItemSelectorCriterion extends BaseItemSelectorCriterion {
+
+	@Override
+	public String getMimeTypeType() {
+		return ItemSelectorCriterionConstants.MIME_TYPE_IMAGE_TYPE;
+	}
```
Comment (adolfopa) [inline modules/apps/blogs/blogs-item-selector-api/src/main/java/com/liferay/blogs/item/selector/criterion/BlogsItemSelectorCriterion.java:28]: Would it be ok rename this method to `getMimeTypeRestriction`?

File: modules/apps/blogs/blogs-item-selector-web/src/main/java/com/liferay/blogs/item/selector/web/internal/display/context/BlogsItemSelectorViewDisplayContext.java [hunk 2]
```diff
@@ -89,6 +89,10 @@ public PortletURL getEditImageURL(
 		).buildPortletURL();
 	}
 
+	public String getFileType() {
+		return _blogsItemSelectorCriterion.getMimeTypeType();
+	}
+
```
Comment (adolfopa) [inline modules/apps/blogs/blogs-item-selector-web/src/main/java/com/liferay/blogs/item/selector/web/internal/display/context/BlogsItemSelectorViewDisplayContext.java:95]: Same here: `getFileType` -> `getMimeTypeRestriction`.

File: modules/apps/blogs/blogs-item-selector-web/src/main/resources/META-INF/resources/blogs_attachments.jsp [hunk 3]
```diff
@@ -75,6 +75,7 @@ if (folder != null) {
 	editImageURL="<%= blogsItemSelectorViewDisplayContext.getEditImageURL(liferayPortletResponse) %>"
 	emptyResultsMessage='<%= LanguageUtil.get(resourceBundle, "there-are-no-blog-attachments") %>'
 	extensions="<%= ListUtil.fromArray(blogsItemSelectorViewDisplayContext.getImageExtensions()) %>"
+	fileType="<%= blogsItemSelectorViewDisplayContext.getFileType() %>"
```
Comment (adolfopa) [inline modules/apps/blogs/blogs-item-selector-web/src/main/resources/META-INF/resources/blogs_attachments.jsp:78]: fileType -> mimeTypeRestriction

File: modules/apps/item-selector/item-selector-api/src/main/java/com/liferay/item/selector/constants/ItemSelectorCriterionConstants.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ */
+public class ItemSelectorCriterionConstants {
+
+	public static final String MIME_TYPE_DEFAULT_TYPE = StringPool.STAR;
+
+	public static final String MIME_TYPE_IMAGE_TYPE = "image";
+
+	public static final String MIME_TYPE_VIDEO_TYPE = "video";
+
+}
```
Comment (adolfopa) [inline modules/apps/item-selector/item-selector-api/src/main/java/com/liferay/item/selector/constants/ItemSelectorCriterionConstants.java:30]: ItemSelectorCriterionConstants -> MimeTypeRestrictionConstants.
MIME_TYPE_VIDEO_TYPE -> MIME_TYPE_RESTRICTION_VIDEO

**Review by adolfopa — CHANGES_REQUESTED** [review]
Looks good. I've suggested some naming changes; that would need to be changed in a lot of places, but I've only commented in a couple of them. Let me know if you need more details.

**Review by ambrinchaudhary — APPROVED** [review]
Reviewed frontend part, LGTM!

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
ci:test:sf

File: modules/apps/item-selector/item-selector-api/src/main/java/com/liferay/item/selector/ItemSelectorCriterion.java [hunk 5]
```diff
@@ -60,6 +62,10 @@
 	 */
 	public List<ItemSelectorReturnType> getDesiredItemSelectorReturnTypes();
 
+	public default String getMimeTypeRestriction() {
+		return StringPool.STAR;
```
Comment (adolfopa) [inline modules/apps/item-selector/item-selector-api/src/main/java/com/liferay/item/selector/ItemSelectorCriterion.java:66]: Why not use here the `MIME_TYPE_RESTRICTION_DEFAULT` constant?

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 6]
```diff
@@ -5243,11 +5243,13 @@ draft-version=Draft Version
...
 drafts=Drafts
 drag-and-drop-criterion-from-the-right-to-add-rules=Drag and drop criterion from the right to add rules.
-drag-and-drop-or-click-to-upload=Drag & Drop or Click to Upload
 drag-and-drop-over-an-existing-criteria-to-form-groups=Drag and drop over an existing criteria to form groups.
 drag-and-drop-to-upload=Drag and Drop to Upload
 drag-and-drop-to-upload-or-x=<strong>Drag</strong> & <strong>Drop</strong> to <strong>Upload</strong> or {0}
 drag-and-drop-to-upload-to-x-or=<strong>Drag</strong> & <strong>Drop</strong> to <strong>Upload</strong> to {0} or
+drag-and-drop-your-files-or-click-to-upload=Drag & Drop your files or Click here to Upload
+drag-and-drop-your-images-or-click-to-upload=Drag & Drop your images or Click here to Upload
+drag-and-drop-your-videos-or-click-to-upload=Drag & Drop your videos or Click here to Upload
```
Comment (adolfopa) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:5252]: Capitalization is wrong. Use https://capitalizemytitle.com/ to fix it.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/item-selector/item-selector-api/src/main/java/com/liferay/item/selector/ItemSelectorCriterion.java (hunk 5 above)
Comment (boton) [reply modules/apps/item-selector/item-selector-api/src/main/java/com/liferay/item/selector/ItemSelectorCriterion.java:66]: I will, I forgot on this case

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 6 above)
Comment (boton) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:5252]: I directly copy the design text, I will fix it

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:reevaluate:1257401_2246

**Pull request closed without merging**